.pre-commit-config.yaml

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+ci:
+    autofix_commit_msg: |
+      [pre-commit.ci] auto code formatting
+    autofix_prs: true
+    autoupdate_branch: ''
+    autoupdate_commit_msg: '[pre-commit.ci] pre-commit autoupdate'
+    autoupdate_schedule: quarterly
+    skip: []
+    submodules: false
+
+repos:
+  - repo: https://github.com/pre-commit/mirrors-clang-format
+    rev: v17.0.6
```

Comment:
Just to be sure, that version is the one we have to keep in synch with the devcontainers?

Would it be possible to create a github action that checks whether the pre-commit-config is up to date 🤔

Reply:
> Would it be possible to create a github action that checks whether the pre-commit-config is up to date

Theoretically yes, but it would be more effort than we want to ask of @bdice right now. I don't envision this version changing very often, so we'll live with it being manual for now.

Reply:
I checked `clang-format --version` and it reported 17.0.6. I suppose it would be possible to automate this but I think that could be evaluated at a later time depending on how much headache there is. I would recommend that we get past the point of applying clang-format across the codebase before worrying about version management.

Reply:
Yeah it was more me thinking out loud